File: CS201/Assignments/PythonJavaAll/LineReader.java
import java.io.File;
import java.util.Scanner;
import java.io.FileNotFoundException;

/**
 * LineReader.java
 * Jeff Ondich, 2014-01-01
 * Modified: Anna Rafferty, 2016-08-24
 * Modified: Layla Oesper, 2017-08-30
 *
 * A very brief introduction to File and Scanner, excessively commented
 * for tutorial purposes.
 *
 * This is the Java half of a pair of parallel examples in Python and Java.
 * See linereader.py.
 *
 * Try a few things:
 *
 * 1. Create a small text file called somefile.txt, put it in the same
 *    directory as LineReader.java, and then compile and run LineReader.
 *    ("javac LineReader.java" followed by "java LineReader")
 *
 * 2. What happens when you change the inputFilePath variable to "nonexistent.txt"?
 *
 * 3. What happens when you remove the "= null" from the Scanner declaration?
 *
 * 4. What happens when you remove the try/catch blocks and just do
 *    "Scanner scanner = new Scanner(inputFile)"?
 *
 * 5. Visit the java String documentation and play around with some String
 *    methods other than toUpperCase.
 */
public class LineReader {
    public static void main(String[] args) {
        // Create a File object connected to the file you want to read.
        String inputFilePath = "somefile.txt";
        File inputFile = new File(inputFilePath);

        // Create a Scanner object connected to your file. This is where
        // the JVM tries to actually open the file, and thus this is
        // where an exception can occur. That's why there's a try/catch
        // block to catch the exception. It allows the programmer to
		// build in a way for the program to recover from the error (in
		// this case, by just printing the error for the user).
        Scanner scanner = null;
        try {
            scanner = new Scanner(inputFile);
        } catch (FileNotFoundException e) {
            System.err.println(e);
            System.exit(1);
        }

        // Get one line at a time from the file, and print each line in upper
        // case to standard output.
        int numberOfLines = 0;
        while (scanner.hasNextLine()) {
            String line = scanner.nextLine();
            System.out.println(line.toUpperCase());
            numberOfLines++;
        }
        scanner.close();
        System.out.println("\nNumber of lines in file: " + numberOfLines);

    }
}

File: CS201/Assignments/PythonJavaAll/linereader.py
'''
   linereader.py
   Jeff Ondich, 2013-01-04
   Anna Rafferty, 2016-08-24 (conversion to python3)
   Layla Oesper, 2017-08-30

   Reads lines from a file, prints them in upper-case, and prints
   a count of the lines at the end.
   
   Intended as the Python half of parallel examples in Python and
   Java. See LineReader.java.
   
   Try a few things:
 
   1. Create a small text file called somefile.txt, put it in the same
      directory as linereader.py, and then run linereader.py.
      ("python3 linereader.py")
  
   2. What happens when you change the inputFilePath variable to 'nonexistent.txt'?
   
'''
import sys

inputFilePath = 'somefile.txt'
try:
    inputFile = open(inputFilePath)
except Exception as e:
    print('Cannot open', inputFilePath, file=sys.stderr)
    exit()

numberOfLines = 0
for line in inputFile:
    print(line.upper(), end='')
    numberOfLines += 1
print('\nNumber of lines in file:', numberOfLines)